lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -71,8 +76,21 @@ def relationship_value_for(association, options)
         return association.options[:virtual_value] if association.options[:virtual_value]
         return unless association.serializer && association.serializer.object
 
+        if options[:fields].is_a? Array
+          fields = options[:fields].select do |k| #I would prefer using {...} here but RuboCop fails :/
+            k.is_a? Hash
+          end.try(:first).try(:[], association.name)
+        end
+
         opts = instance_options.merge(include: @include_tree[association.key])
-        Attributes.new(association.serializer, opts).serializable_hash(options)
+        options = options.merge(fields: fields)
+
+        Attributes.new(
+          association.serializer,
+          opts.merge(association.options)
+        ).serializable_hash(
```

Comment:
we need to rename internal apis like `serializable_hash` here to `as_json`. it's just confusing when it's not part of the public serialization api. e.g. https://github.com/rails-api/active_model_serializers/issues/1643